tests/t0123_gateway_json_cbor_test.go

```diff
@@ -1,13 +1,859 @@
+					IsJSONEqual([]byte(`{"hello": "this is not a link"}`)),
+				),
+		},
+	}
+
+	test.Run(t, tests.Build())
+}
+
+// ## NATIVE TESTS for DAG-JSON (0x0129) and DAG-CBOR (0x71):
+// ## DAG- regression tests for core behaviors when native DAG-(CBOR|JSON) is requested
+func TestNativeDag(t *testing.T) {
+	missingCID := car.RandomCID()
+
+	table := []struct {
+		Name        string
+		Format      string
+		Disposition string
+		Checker     func(value []byte) Check[[]byte]
+	}{
+		{"plain JSON codec", "json", "inline", IsJSONEqual},
+		{"plain CBOR codec", "cbor", "attachment", IsEqualBytes},
+	}
+
+	for _, row := range table {
+		dagTraversal := car.MustOpenUnixfsCar(fmt.Sprintf("t0123/dag-%s-traversal.car", row.Format)).MustGetRoot()
+		dagTraversalCID := dagTraversal.Cid()
+		formatted := dagTraversal.Formatted("dag-" + row.Format)
+
+		tests := SugarTests{
+			/**
+			  # GET without explicit format and Accept: text/html returns raw block
+
+			  test_expect_success "GET $name from /ipfs without explicit format returns the same payload as the raw block" '
+			  ipfs block get "/ipfs/$CID" > expected &&
+			  curl -sX GET "http://127.0.0.1:$GWAY_PORT/ipfs/$CID" -o curl_output &&
+			  test_cmp expected curl_output
+			  '
+			*/
+			{
+				Name: fmt.Sprintf("GET %s from /ipfs without explicit format returns the same payload as the raw block", row.Name),
+				Hint: `GET without explicit format and Accept: text/html returns raw block`,
+				Request: Request().
+					Path("ipfs/%s", dagTraversalCID),
+				Response: Expect().
+					Status(200).
+					Body(
+						row.Checker(formatted),
+					),
+			},
+			/**
+			  # GET dag-cbor block via Accept and ?format and ensure both are the same as `ipfs block get` output
+
+			  test_expect_success "GET $name from /ipfs with format=dag-$format returns the same payload as the raw block" '
+			  ipfs block get "/ipfs/$CID" > expected &&
+			  curl -sX GET "http://127.0.0.1:$GWAY_PORT/ipfs/$CID?format=dag-$format" -o curl_ipfs_dag_param_output &&
+			  test_cmp expected curl_ipfs_dag_param_output
+			  '
+			*/
+			{
+				Name: fmt.Sprintf("GET %s from /ipfs with format=dag-%s returns the same payload as the raw block", row.Name, row.Format),
+				Hint: `GET dag-cbor block via Accept and ?format and ensure both are the same as ipfs block get output`,
+				Request: Request().
+					Path("ipfs/%s", dagTraversalCID).
+					Query("format", fmt.Sprintf("dag-%s", row.Format)),
+				Response: Expect().
+					Status(200).
+					Body(
+						row.Checker(formatted),
+					),
+			},
+			/**
+			  test_expect_success "GET $name from /ipfs for application/$format returns the same payload as format=dag-$format" '
+			  curl -sX GET "http://127.0.0.1:$GWAY_PORT/ipfs/$CID?format=dag-$format" -o expected &&
+			  curl -sX GET "http://127.0.0.1:$GWAY_PORT/ipfs/$CID?format=$format" -o plain_output &&
+			  test_cmp expected plain_output
+			  '
+			*/
+			// TODO(lidel): Note we disable this test, we check the payloads above.
```

Comment:
@lidel a note here, we ignore a few tests: they compare two requests with each other.
Instead, we test each request.